docs/visualizations.md

```diff
@@ -0,0 +1,57 @@
+# Visualization workflow
+
+## Purpose of visualizations
+Visualizations have two main purposes:
+
+- Display results of computations. Each node can be assigned with a visualization (displayed next to the node). After a node computes its new value, the visualization shows it in an understandable way to the user.
+
+- Control the computations. Think about visualizations as a separate component, just as nodes, with their own input and output ports. Whenever new data appears on the input port, the visualization displays it. However, when the input port is not connected, the visualization behaves like a data-producing widget. For example, a map visualization can display locations encoded in its input data. It can also allow us to interactively pick new locations with the mouse when no input is connected and sends the manually selected locations to the output port. Similarly, the histogram can be used to manually draw a histogram with the mouse to set data on the next nodes. A number visualization is, at the same time, a slider. Image visualizations can behave like an image editor, etc.
+
+
+## Visualization Display Forms
+Visualizations can be displayed in the following ways:
+
+- **Attached to nodes**  
+  Their input and output ports are not visible in this form. They are used only to display the result of the attached node. Whenever you move the node, the visualization moves as well. The visualization can be shown this way by selecting nodes by clicking the "space" button.
+  
+- **Fullscreen**  
+  Similar to the previous mode, but the visualization occupies the full visual space. This form can be triggered on a recently selected node (even if many nodes are selected, we remember the last selected one) by either pressing space and keeping it pressed for longer than approx 0.5 s, or by clicking space twice. In the former case, the visualization stops being fullscreen whenever we release space, in the later, whenever we click space again.
```

Comment:
```suggestion
  Similar to the previous mode, but the visualization occupies the full visual space. This form can be triggered on a recently selected node (even if many nodes are selected, we remember the last selected one) by either pressing space and keeping it pressed for longer than approx 0.5 s, or by tapping space bar twice. In the former case, the visualization stops being fullscreen whenever we release space, in the later, whenever we press space again.
```

Reply:
One more thing: full visual space - do you mean the full screen or IDE space?

Reply:
of course IDE